spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -1415,21 +1415,23 @@ def _do_run(self, n_machine_time_steps, loading_done):
                     self._load_outputs["ExecutableTargets"],
                     self._executable_finder)
 
-        # update algorithm list with extra pre algorithms if needed
-        if self._extra_pre_run_algorithms is not None:
-            algorithms = list(self._extra_pre_run_algorithms)
-        else:
-            algorithms = list()
+        algorithms = list()
 
         # If we have run before, make sure to extract the data before the next
         # run
         if (self._has_ran and not self._has_reset_last and
                 not self._use_virtual_board):
             algorithms.append("FrontEndCommonBufferExtractor")
 
-            # check if we need to clear the iobuf during runs
-            if self._config.getboolean("Reports", "clear_iobuf_during_run"):
-                algorithms.append("FrontEndCommonChipIOBufClearer")
+        # update algorithm list with extra pre algorithms if needed
+        if self._extra_pre_run_algorithms is not None:
+            algorithms = list(self._extra_pre_run_algorithms)
```

Comment:
I can't trust that that code will work; we don't want to wipe out the algorithm we potentially added above. I believe that line 1428 should instead be:
```python
algorithms.extend(self._extra_pre_run_algorithms)
```